wasm/README.md

```diff
@@ -0,0 +1,42 @@
+# How to optimize your binary size?
+
+Smartcontracts should be optimized for because they are stored directly on the blockchain. By
+following this list of optimization steps your binary's size can be reduced by an order of magnitude
+(e.g. from 1.1MB to 100KB):
+
+1. Create a `Cargo.toml` akin to the following template:
+
+```toml
+  [package]
+  name = "smartcontract"
+  version = "0.1.0"
+  edition = "2021"
+
+  [lib]
+  crate-type = ['cdylib']    # Crate has to be linked dynamically
```

Comment:
Add explanation. 

A user knows that Rust is statically linked by default because of unstable ABI, meaning that binaries compiled by different compilers might not be compatible. 

The reason why we want dynamic linking, is because we will store wasm VMs compatible with all binary versions anyway, so the dynamic libraries will always be available. This is a contract that we should make explicitly with the developers of smartcontracts.